Source/Python/WarpXWrappers.cpp

```diff
@@ -652,6 +652,30 @@ namespace
         particle_buffers.clearParticles();
     }
 
+    void warpx_depositChargeDensity (const char* char_species_name, int lev) {
+        // this function is used to deposit a given species' charge density
+        // in the rho_fp multifab which can then be accessed from python via
+        // pywarpx.fields.RhoFPWrapper()
+        WarpX& warpx = WarpX::GetInstance();
+        const auto & mypc = warpx.GetPartContainer();
+        const std::string species_name(char_species_name);
+        auto & myspc = mypc.GetParticleContainerFromName(species_name);
+        auto rho_fp = warpx.get_pointer_rho_fp(lev);
```

Comment:
This should check that `rho_fp` is not null, since it will not alway be allocated.